src/main/java/org/apache/maven/plugins/assembly/filter/ComponentsXmlArchiverFileFilter.java

```diff
@@ -102,7 +102,7 @@ private void addToArchive( final Archiver archiver )
     {
         if ( components != null )
         {
-            final File f = File.createTempFile( "maven-assembly-plugin", "tmp" );
```

Comment:
ok, content of META-INF/plexus/components.xml can be read when merging: nothing sensitive